proxy/proxy.go

```diff
@@ -115,630 +116,275 @@ func (p *Server) isStopped() bool {
-		p.writeBlockedResponse(w, r)
-		return
-	}
-
-	// Forward regular HTTP request
-	p.forwardRequest(w, r, false)
-}
-
-// forwardRequest forwards a regular HTTP request
-func (p *Server) forwardRequest(w http.ResponseWriter, r *http.Request, https bool) {
-	p.logger.Debug("forwardHTTPRequest called", "method", r.Method, "url", r.URL.String(), "host", r.Host)
-
-	s := "http"
-	if https {
-		s = "https"
-	}
-	// Create a new request to the target server
-	targetURL := &url.URL{
-		Scheme:   s,
-		Host:     r.Host,
-		Path:     r.URL.Path,
-		RawQuery: r.URL.RawQuery,
-	}
-
-	p.logger.Debug("Target URL constructed", "target", targetURL.String())
-
-	// Create HTTP client with very short timeout for debugging
-	client := &http.Client{
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse // Don't follow redirects
-		},
-	}
-
-	// Create new request
-	req, err := http.NewRequest(r.Method, targetURL.String(), r.Body)
+func (p *Server) handleConnectionWithTLSDetection(conn net.Conn) {
+	// Detect protocol using TLS handshake detection
+	wrappedConn, isTLS, err := p.isTLSConnection(conn)
 	if err != nil {
-		p.logger.Error("Failed to create forward request", "error", err)
-		http.Error(w, fmt.Sprintf("Failed to create request: %v", err), http.StatusInternalServerError)
-		return
-	}
+		p.logger.Error("Failed to check connection type", "error", err)
 
-	// Copy headers
-	for name, values := range r.Header {
-		// Skip connection-specific headers
-		if strings.ToLower(name) == "connection" || strings.ToLower(name) == "proxy-connection" {
-			continue
-		}
-		for _, value := range values {
-			req.Header.Add(name, value)
+		err := conn.Close()
+		if err != nil {
+			p.logger.Error("Failed to close connection", "error", err)
 		}
-	}
-
-	p.logger.Debug("About to make HTTP request", "target", targetURL.String())
-	resp, err := client.Do(req)
-	if err != nil {
-		p.logger.Error("Failed to make forward request", "error", err, "target", targetURL.String(), "error_type", fmt.Sprintf("%T", err))
-		http.Error(w, fmt.Sprintf("Failed to make request: %v", err), http.StatusBadGateway)
 		return
 	}
-	defer func() { _ = resp.Body.Close() }()
-
-	p.logger.Debug("Received response", "status", resp.StatusCode, "target", targetURL.String())
-
-	// Copy response headers (except connection-specific ones)
-	for name, values := range resp.Header {
-		if strings.ToLower(name) == "connection" || strings.ToLower(name) == "transfer-encoding" {
-			continue
-		}
-		for _, value := range values {
-			w.Header().Add(name, value)
-		}
-	}
-
-	// Copy status code
-	w.WriteHeader(resp.StatusCode)
-
-	// Copy response body
-	bytesWritten, copyErr := io.Copy(w, resp.Body)
-	if copyErr != nil {
-		p.logger.Error("Error copying response body", "error", copyErr, "bytes_written", bytesWritten)
-		http.Error(w, "Failed to copy response", http.StatusBadGateway)
+	if isTLS {
+		p.logger.Debug("🔒 Detected TLS connection - handling as HTTPS")
```

Comment:
nit: I think we should avoid usage of emoji in logging

Reply:
will do in a follow-up PR